views/partials/head.ejs

```diff
@@ -4,11 +4,43 @@
 <meta http-equiv='x-ua-compatible' content='ie=edge'>
 <link rel='stylesheet' href='/assets/css/bootstrap.min.css'>
 <style>
-body {padding-top: 20px;}
-a pre {color: inherit;}
-.preformatted {white-space: pre;}
-/* go away, LastPass */
-form[autocomplete="off"] input[type="text"] {
-  background-position: 150% 50% !important;
-}
+  body {
+    padding-top: 20px;
+  }
+
+  a pre {
+    color: inherit;
+  }
+
+  .preformatted {
+    white-space: pre;
+  }
+
+  body {
+    padding-top: 20px;
+  }
+
+  /* for the sticky nav to follow the user all the way down the page, all parent elements must be full height too */
+
+  html,
+  body,
+  main {
+    min-height: 100vh;
+  }
+
+  nav {
+    position: sticky;
+    top: 0;
+    z-index: 100
```

Comment:
It doesn't look very nice when buttons overlay editor area on scrolling.
I haven't looked into it to see how to do it better but I believe better solution would be to use flexbox to stretch everything to full window height and then make editor area scrollable.

Reply:
Thank-you for all the feedback @rchl and @aaronshaf! I'm interpreting this as these features are valuable to you as well, and worth merging after the code is improved.

I'll have to put some thought into this point. I agree this doesn't look nice, but I found the native browser search does not search text that CodeMirror and Ace isn't rendering because it is out of view. Making the editor render all text by expanding the height to display all content fixed this for me, and then I wanted sticky positioning for buttons to stop me from scrolling up to save with a click

Three things I can try to address this:

1. Improve the appearance of the sticky buttons. I could place them on a 100% width white background with some margin that sticks with them as the user scrolls. That will look more normal.

2. Try out CodeMirror's native search feature and see if it feels as good as the native browser search. I expect it will search through offscreen content, so sticky positioning would not be needed.

3. Try out CodeMirror's `viewMargin: Infinity` feature which should fix native browser search with offscreen content (https://codemirror.net/doc/manual.html#option_viewportMargin). I don't think the performance will be worse than it is now in this branch since 100% of the JSON document is being rendered already. Sticky positioning would not be needed with this option either.